compiler/semstmts.nim

```diff
@@ -740,7 +741,17 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
         vm.setupCompileTimeVar(c.module, c.idgen, c.graph, x)
       if v.flags * {sfGlobal, sfThread} == {sfGlobal}:
         message(c.config, v.info, hintGlobalVar)
-
+      let isGlobal = {sfGlobal, sfPure} <= v.flags
+      if isGlobal and def.kind == nkCall:
```

Comment:
Right idea but move it to a helper proc and make it recursive so that e.g. `f(g(localVar))` is caught too.